core/src/main/scala/org/http4s/grpc/ServerGrpc.scala

```diff
@@ -105,24 +107,25 @@ object ServerGrpc {
   ): HttpRoutes[F] = HttpRoutes.of[F]{
     case req@POST -> Root / sN / mN if sN === serviceName && mN === methodName =>
       for {
-        status <- Ref.of[F, Int](0)
-        trailers = status.get.map(i =>
+        status <- Ref.of[F, (Int, Option[String])]((0, Option.empty))
+        trailers = status.get.map{case (i, message) =>
           Headers(
             NamedHeaders.GrpcStatus(i),
-          )
-        )
+          ).put(message.map(NamedHeaders.GrpcMessage(_)))
+        }
         timeout = req.headers.get[NamedHeaders.GrpcTimeout]
 
       } yield {
         val body = Stream.eval(f(codecs.Messages.decode(decode)(req.body), req.headers))
           .flatMap(codecs.Messages.encodeSingle(encode)(_))
           .through(timeoutStream(_)(timeout.map(_.duration)))
           .onFinalizeCase{
-            case Resource.ExitCase.Errored(_: TimeoutException) => status.set(4)
-            case Resource.ExitCase.Errored(_) => status.set(2)
-            case Resource.ExitCase.Canceled => status.set(1)
+            case Resource.ExitCase.Errored(_: TimeoutException) => status.set((4, None))
+            case Resource.ExitCase.Errored(e) => status.set((2, e.getMessage().some))
```

Comment:
I wonder if `toString` is better, that way it also includes the class of the exception in addition to the message. I think :)

```suggestion
            case Resource.ExitCase.Errored(e) => status.set((2, e.toString().some))
```

Reply:
I went with what fs2-grpc was returning. I dont mind a different choice though.